internal/services/mssql/mssql_managed_instance_resource.go

```diff
@@ -0,0 +1,538 @@
+	return validate.ManagedInstanceID
+}
+
+func (r MsSqlManagedInstanceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ValidateMsSqlServerName,
+		},
+
+		"location": azure.SchemaLocation(),
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"sku_name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"GP_Gen4",
+				"GP_Gen5",
+				"BC_Gen4",
+				"BC_Gen5",
+			}, false),
+		},
+
+		"administrator_login": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"administrator_login_password": {
+			Type:         schema.TypeString,
+			Required:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"vcores": {
+			Type:     schema.TypeInt,
+			Required: true,
+			ValidateFunc: validation.IntInSlice([]int{
+				4,
+				8,
+				16,
+				24,
+				32,
+				40,
+				64,
+				80,
+			}),
+		},
+
+		"storage_size_in_gb": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntBetween(32, 8192),
+		},
+
+		"license_type": {
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"LicenseIncluded",
+				"BasePrice",
+			}, true),
+		},
+
+		"subnet_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
+
+		"collation": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      "SQL_Latin1_General_CP1_CI_AS",
+			ValidateFunc: validation.StringIsNotEmpty,
+			ForceNew:     true,
+		},
+
+		"public_data_endpoint_enabled": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"minimum_tls_version": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  "1.2",
+			ValidateFunc: validation.StringInSlice([]string{
+				"1.0",
+				"1.1",
+				"1.2",
+			}, false),
+		},
+
+		"proxy_override": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  string(sql.ManagedInstanceProxyOverrideDefault),
+			ValidateFunc: validation.StringInSlice([]string{
+				string(sql.ManagedInstanceProxyOverrideDefault),
+				string(sql.ManagedInstanceProxyOverrideRedirect),
+				string(sql.ManagedInstanceProxyOverrideProxy),
+			}, false),
+		},
+
+		"timezone_id": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Default:      "UTC",
+			ValidateFunc: validation.StringIsNotEmpty,
+			ForceNew:     true,
+		},
+
+		"dns_zone_partner_id": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
+
+		"identity": identity.SystemAssigned{}.Schema(),
```

Comment:
can we update this to use the Common Schema and ExpandSystemAssigned / FlattenSystemAssigned packages from `go-azure-helpers`? I'll be removing this package shortly